### PR TITLE
Fix release build with WalletCore

### PR DESCRIPTION
**Pull request by darinwf**
Trying to get the release build to work with the new WalletCore framework.

This branch is not a complete solution. Still seeing this error when trying to upload to app store connect:

```
Uploading build 1.31.0 to Testflight...
*** Error: Error uploading '/tmp/Neeva.ipa'.
*** Error: This bundle is invalid. The application-identifier entitlement is missing; it should contain your 10-character Apple Developer ID, followed by a dot, followed by your bundle identifier. With error code STATE_ERROR.VALIDATION_ERROR.90075 for id b1aa1ebc-63d4-454b-8c51-bc11a562121c Asset validation failed (-19208)
 {
    NSLocalizedDescription = "This bundle is invalid. The application-identifier entitlement is missing; it should contain your 10-character Apple Developer ID, followed by a dot, followed by your bundle identifier. With error code STATE_ERROR.VALIDATION_ERROR.90075 for id b1aa1ebc-63d4-454b-8c51-bc11a562121c";
    NSLocalizedFailureReason = "Asset validation failed";
}
```

@chung-lai - any idea what might be needed?

### CODE REVIEW

**Comment by darinwf** [comment]
Looking at this:
https://developer.apple.com/library/archive/qa/qa1710/_index.html

**Comment by chung-lai** [comment]
Could it be the bundle identifier for `WalletCore` is not set properly? When I open the project file, it complains about this:

Screen Shot 2022-03-03 at 1 08 28 PM

For `Shared`, the bundle identifier is set to: `co.neeva.app.ios.browser.Shared` but when I look at `WalletCore`, it's `com.neeva.WalletCore`. We should probably rename it to something like `co.neeva.app.ios.browser.WalletCore`

**Comment by darinwf** [comment]
@chung-lai - Yeah, I made that change on this branch. I'm going to try setting `application-identifier` directly and see if that helps. Following this: https://stackoverflow.com/questions/28999820/error-itms-90075-the-application-identifier-entitlement-is-missing#28999821

**Comment by darinwf** [comment]
Oh, I got a better clue this time:
```
Package Summary:
 
1 package(s) were not uploaded because they had problems:
	/var/folders/7p/l24_t0yd6jjfs8j74mvrjqc00000gn/T/7BA86872-76CC-4561-8819-8A1F5E0E58BA/e7c3a76027a3304c27c3bfc34b248cc6.itmsp - Error Messages:
		ERROR ITMS-90164: "Invalid Code Signing Entitlements. The entitlements in your app bundle signature do not match the ones that are contained in the provisioning profile. According to the provisioning profile, the bundle contains a key value that is not allowed: 'A9735M6NPH.co.neeva.app.ios.browser.WalletCore' for the key 'application-identifier' in 'Payload/Client.app/Frameworks/WalletCore.framework/WalletCore'."

*** Error: Error uploading '/tmp/Neeva.ipa'.
*** Error: ERROR ITMS-90164: "Invalid Code Signing Entitlements. The entitlements in your app bundle signature do not match the ones that are contained in the provisioning profile. According to the provisioning profile, the bundle contains a key value that is not allowed: 'A9735M6NPH.co.neeva.app.ios.browser.WalletCore' for the key 'application-identifier' in 'Payload/Client.app/Frameworks/WalletCore.framework/WalletCore'." (-18000)
```

**Comment by darinwf** [comment]
OK, found a solution.

Important to set `CODE_SIGN_IDENTITY = "";` for WalletCore just like Shared.

File: Client/Entitlements/Neeva.entitlements [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <plist version="1.0">
 <dict>
 	<key>aps-environment</key>
-	<string>development</string>
+	<string>production</string>
```
Comment (darinwf) [inline Client/Entitlements/Neeva.entitlements:6]: @chung-lai - I don't quite understand why this was necessary or why it wasn't necessary before. Hope this doesn't screw something else up.

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6596,10 +6577,10 @@
 				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
-				CODE_SIGN_IDENTITY = "iPhone Distribution";
-				CODE_SIGN_STYLE = Automatic;
+				CODE_SIGN_IDENTITY = "";
```
Comment (darinwf) [inline Client.xcodeproj/project.pbxproj:6580]: Setting `CODE_SIGN_IDENTITY` to empty here turned out to be critical. This is what we have for `Shared`. The IDE doesn't let you set this value, so I had to hand-edit it to be this way. At least it works!

File: Client.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1170,17 +1169,6 @@
 			name = "Copy Files";
 			runOnlyForDeploymentPostprocessing = 0;
 		};
-		4A8CE5DC27CCA0A600735C75 /* Embed Frameworks */ = {
```
Comment (darinwf) [inline Client.xcodeproj/project.pbxproj:1173]: Important not to embed Shared twice in the final IPA

File: Client/Entitlements/Neeva.entitlements (hunk 1 above)
Comment (chung-lai) [reply Client/Entitlements/Neeva.entitlements:6]: Interesting, is this required for the build upload to work? This setting seem to have something to do with the push notification service but `development` (i believe it's the sandbox push notification server) is meant for debug build.

I don't see harm doing this or I can try it out later see if my debug build and Testflight still working the way as expected.

**Review by chung-lai — APPROVED** [review]

**Pull request merged**